examples/block-explorer/explorer-indexer/Cargo.toml

```diff
@@ -18,3 +18,4 @@ fuels-types = { version = "0.38.1", default-features = false }
 getrandom = { version = "0.2", features = ["js"] }
 serde = { version = "1.0", default-features = false, features = ["derive"] }
 serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
+sha2 = { version = "0.10" }
```

Comment:
@deekerno It's not clear what this is used for? It's not used in the user's indexer (because I don't see a change to the explorer indexer library) so I'm imagining this was needed by the codgen? If so, this should go through the plugin preludes

Reply:
It is indeed required by the codegen. Now that you mention it, I had originally put it in the prelude, but it looks like it got blown away through a rebase and I forgot about it. In any case, wouldn't it still need to be in the dependencies as the calculation happens when the receipt arrives and the hashing comes from a crate that isn't a part of the standard library?

Reply:
- So the only dependencies that go in an indexer's Cargo.toml are dependencies used in the user-facing part of the indexer (the codegen not being a user-facing part of the indexer because the user has no control over it outside of their own indexer handlers).
  - If I was calculating some sha2 of some byte array and creating an ID from that and assigning that ID to one of my GraphQL types - that's a case where I (as the indexer creator) would import sha2 in my indexer

- When you add a piece of code to the codegen that references an outside dependency (e.g., sha2), you add it to the plugin's Cargo.toml, then make it available via plugin's prelude (wasm and native) -- (e.g., `pub use sha2;` ).
  - This only makes the dependency available to be used by the macros - you still have to import it the dependency into the macros though

- To import this dependency from the plugin, into the indexer macros, you'll then add the dependency to the macros prelude. So the flow is (1) add dependency to plugin Cargo.toml (2) export dependency in plugin's prelude (3) import dependency in _macros_ prelude -- at which point it is available in the indexer macro now.

- [I see that there is already a sha2 available via the plugin in your PR](https://github.com/FuelLabs/fuel-indexer/blob/master/packages/fuel-indexer-macros/src/wasm.rs#L42) as direct import. You don't want to import dependencies directly into the macros. We want to add that dependency to the _plugin_, then import that dependency from the plugin, into the macros (hope that makes sense).

CC @rvmelkonian cause that's ☝🏽 important

Reply:
Yeah, that makes sense; I had understood it differently. I'll make the change in a follow-up PR.